content/blog/2022-06-01-Jupyter-in-cloud-with-TPI.md

```diff
@@ -0,0 +1,379 @@
+---
+title: Jupyter Notebooks in any cloud with TPI
+date: 2022-06-01
+description: >
+  Use TPI to quickly and cheaply provision a cloud-hosted Jupyter server to run
+  your notebooks on.
+descriptionLong: >
+  We can use TPI to quickly and cheaply provision cloud instances. This guide
+  will explore how to use TPI to launch Jupyter Server on an Amazon Web Services
+  instance at the lowest possible cost. With minor modifications, you can also
+  launch instances on Google Cloud, Microsoft Azure, or Kubernetes.
+picture: 2022-06-01/tpi-jupyter-cover.jpeg
+author: rob_dewit
+# commentsUrl: TODO
+tags:
+  - TPI
+  - Jupyter Notebook
+  - Cloud
+  - AWS
+---
+
+<admon type="info">
+
+I based this blog post on
+[@casperdcl's guide published over on Hackernoon](https://hackernoon.com/using-jupytertensorboard-in-any-cloud-with-one-command).
+
+</admon>
+
+Jupyter Notebook is one of —if not the— most-used tools in data science. It
+provides an intuitive and feature-packed developer experience and, for many
+people, is an indispensable tool when working on machine learning projects.
+
+The most straightforward use is on a local machine. We switch to a virtual
+environment, type `jupyter notebook` in our terminal, and head over to
+http://localhost:8888. We're good to go and ready to start prototyping!
+
+This usage has one major downside, however: we're limited by the specs of our
+local machine. While my laptop won't struggle too much training a classifier for
+the IRIS dataset, it might become a limiting factor when working on an
+alternative to GPT-3. And probably a little while before a project of that
+scope.
+
+The apparent solution to bypassing the limitations of our local machine is to
+move our notebooks to the cloud. When running a notebook on AWS or Azure, we can
+provision as beefy an instance as needed for our project. However, the downside
+to cloud-hosted notebooks is that they're more challenging to get up and
+running. And any time spent setting up our development environment is time we
+can't spend on our actual ML project.
+
+"Now wait a minute," you might say, "hasn't this problem long been solved by
+SaaS solutions like Google Colab and Amazon SageMaker Studio?"
+
+To which I'd reply: "To an extent: yes. But while they certainly have their
+merits, these solutions often have their own limitations which can become
+problematic."
+
+Here are three of the limitations imposed by SaaS solutions such as Colab and
+SageMaker:
+
+- **Cost:** they are expensive to use. You're not only paying for the hardware
+  but also for using the platform.
+- **Walled garden:** they tend to be baked into the vendor's ecosystem. It's
+  difficult and sometimes impossible to use infrastructure components from
+  different vendors in conjunction with each other.
```

Comment:
Have we done any other work to determine what other terminology around this concept is used in the industry.  And re: @casperdcl's Infrastructure components comment, it would be good to insert a "such as ____ and _____" to clarify this.

Reply:
Will insert a phrase with examples.

With regards to `walled garden`: I see the term come up quite often (didn't invent it myself, after all). It's difficult to assess how ubiquitous this is; we can't Google how many people don't know something.

I think that if people don't immediately understand it, they can derive its meaning from context. But we can change the phrase if you think it makes the message clearer.